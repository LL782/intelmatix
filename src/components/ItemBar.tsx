import { Source_Sans_Pro } from "@next/font/google";
import Image from "next/image";
import styles from "@/styles/ItemBar.module.css";

const reg = Source_Sans_Pro({ subsets: ["latin"], weight: "400" });
const bold = Source_Sans_Pro({ subsets: ["latin"], weight: "600" });

export const ItemBar = () => (
  <header className={`${reg.className} ${styles.itemBar}`}>
    <div className={styles.itemSummary}>
      <div className={styles.itemIconSlot}>
        <Image alt="Pickle icon" src="/Pickle.svg" fill />
      </div>
      <h1 className={styles.itemName}>Pickle</h1>
      <p className={`${bold.className} ${styles.itemStock}`}>
        60kg/<span className={styles.maximumStock}>150kg</span>
      </p>
    </div>
    <div className={styles.itemSelectWrapper}>
      <div className={styles.itemSelectedOption}>
        <i className={styles.itemMenuIcon}></i>
        <span className={styles.itemMenuText}>
          <span className={styles.itemMenuName}>Pickle</span> |{" "}
          <span className={styles.itemMenuId}>#1234</span>
        </span>
      </div>
      <select
        className={styles.itemSelect}
        defaultValue="Pickle"
        name="itemSelect"
        id="itemSelect"
      >
        <option value="Pickle">Pickle | #1234</option>
        <option value="Cheese">Cheese | #2345</option>
        <option value="Pattie">Pattie | #3456</option>
        <option value="Bun">Bun | #4567</option>
        <option value="Lettuce">Lettuce | #5678</option>
      </select>
    </div>
  </header>
);
