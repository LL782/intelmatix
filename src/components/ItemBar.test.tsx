import { render, screen } from "@testing-library/react";
import { ItemBar } from "./ItemBar";
import { example } from "@/examples/stockData";

describe("ItemBar", () => {
  describe("Given example data", () => {
    render(<ItemBar data={example} />);
    let header: HTMLElement;

    test("renders a header", () => {
      header = screen.getByRole("header");
      expect(header).toBeDefined();
    });
    test("header contains summary", () => {
      const summary = header.querySelector('[data-testid="itemSummary"]');
      expect(summary).toMatchSnapshot();
    });
    test("header contains item select ", () => {
      const select = header.querySelector('[data-testid="itemSelect"]');
      expect(select).toMatchSnapshot();
    });
  });
});
