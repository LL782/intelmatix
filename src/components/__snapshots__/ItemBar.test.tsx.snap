// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemBar Given example data header contains item select  1`] = `
<div
  class="className itemSelectWrapper"
  data-testid="itemSelect"
>
  <div
    class="itemSelectedOption"
  >
    <i
      class="itemMenuIcon"
    />
    <span
      class="itemMenuText"
    >
      <span
        class="itemMenuName"
      >
        Example
      </span>
       |
       
      <span
        class="itemMenuId"
      >
        #
        0000
      </span>
    </span>
  </div>
  <select
    class="itemSelect"
    id="itemSelect"
    name="itemSelect"
  >
    <option
      selected=""
      value="pickle"
    >
      Pickle
       | #
      1234
    </option>
    <option
      value="cheese"
    >
      Cheese
       | #
      2345
    </option>
    <option
      value="patty"
    >
      Patty
       | #
      3456
    </option>
    <option
      value="bun"
    >
      Bun
       | #
      4567
    </option>
    <option
      value="lettuce"
    >
      Lettuce
       | #
      5678
    </option>
  </select>
</div>
`;

exports[`ItemBar Given example data header contains summary 1`] = `
<div
  class="itemSummary"
  data-testid="itemSummary"
>
  <div
    class="itemIconSlot"
  >
    <img
      alt="Pickle icon"
      data-nimg="fill"
      decoding="async"
      loading="lazy"
      src="/Pickle.svg"
      style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
    />
  </div>
  <h1
    class="className itemName"
  >
    Example
  </h1>
  <p
    class="className itemStock"
  >
    60kg
    /
    <span
      class="maximumStock"
    >
      150kg
    </span>
  </p>
</div>
`;
